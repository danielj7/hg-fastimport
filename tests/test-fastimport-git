#!/bin/sh

# Test a dump of the style created by git fast-export.  Notable features
# of this style:
#   - one single stream
#   - blobs right before the commit that use them
#   - every blob and every commit have a mark
#   - every commit has author and committer (even if they are the same)

# The following dump was created by these commands, with author/
# committer manually munged post-dump:
#   mkdir t && cd t
#   git init
#   echo "hello there" > foo.txt
#   echo "ooga booga" > ooga
#   git add .
#   git commit -m"initial revision"
#   echo "woof woof\nooga booga\nblah blah" > ooga
#   git commit --author="Joe Contributor <joe@example.com>" -a -m"modify"
#   git fast-export --all

. $TESTDIR/fastimport-common

cat > simplegit.dump <<__EOF__
blob
mark :1
data 12
hello there

blob
mark :2
data 11
ooga booga

reset refs/heads/master
commit refs/heads/master
mark :3
author Example <example@example.org> 1200000060 -0400
committer Example <example@example.org> 1200000060 -0400
data 17
initial revision
M 100644 :1 foo.txt
M 100644 :2 ooga

blob
mark :4
data 31
woof woof
ooga booga
blah blah

commit refs/heads/master
mark :5
author Joe Contributor <joe@example.com> 1200000000 -0400
committer Example <example@example.org> 1200000120 -0400
data 7
modify
from :3
M 100644 :4 ooga

__EOF__

echo "% import simple dump from git"
hg init simplegit
hg -R simplegit fastimport simplegit.dump \
  | sed "s|$HGTMP|HGTMP|g"

echo "% hg log"
cd simplegit
hg glog -v

echo "% manifest at rev 0"
hg manifest -r0

echo "% manifest at rev 1"
hg manifest -r1

echo "% cat foo.txt at rev 0"
hg cat -r0 foo.txt
echo "% cat foo.txt at rev 1"
hg cat -r1 foo.txt

echo "% cat ooga at rev 0"
hg cat -r0 ooga
echo "% cat ooga at rev 1"
hg cat -r1 ooga
