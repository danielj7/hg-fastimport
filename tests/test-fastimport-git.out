% import simple dump from git
scanning source...
sorting...
converting...
1 initial revision
0 modify
Removing blob dir 'HGTMP/test-fastimport-git/simplegit/.hg/blobs' ...
% hg log
o  changeset:   1:0a3befda043d
|  tag:         tip
|  user:        Joe Contributor <joe@example.com>
|  date:        Fri Jan 11 01:20:00 2008 +0400
|  files:       ooga
|  description:
|  modify
|
|
o  changeset:   0:9a4b81675bd1
   user:        Example <example@example.org>
   date:        Fri Jan 11 01:21:00 2008 +0400
   files:       foo.txt ooga
   description:
   initial revision


% manifest at rev 0
foo.txt
ooga
% manifest at rev 1
foo.txt
ooga
% cat foo.txt at rev 0
hello there
% cat foo.txt at rev 1
hello there
% cat ooga at rev 0
ooga booga
% cat ooga at rev 1
woof woof
ooga booga
blah blah
